Makefile

```diff
@@ -130,15 +131,18 @@ test-unit-coverage: ##@tests Run unit and integration tests with coverage
 test-e2e: ##@tests Run e2e tests
 	# order: reliability then alphabetical
 	# TODO(tiabc): make a single command out of them adding `-p 1` flag.
-	go test -timeout 5m ./t/e2e/accounts/... -network=$(networkid)
-	go test -timeout 5m ./t/e2e/api/... -network=$(networkid)
-	go test -timeout 5m ./t/e2e/node/... -network=$(networkid)
-	go test -timeout 50m ./t/e2e/jail/... -network=$(networkid)
-	go test -timeout 20m ./t/e2e/rpc/... -network=$(networkid)
-	go test -timeout 20m ./t/e2e/whisper/... -network=$(networkid)
-	go test -timeout 10m ./t/e2e/transactions/... -network=$(networkid)
+	go test -timeout 5m ./t/e2e/accounts/... -network=$(networkid) $(e2e_extraflags)
+	go test -timeout 5m ./t/e2e/api/... -network=$(networkid) $(e2e_extraflags)
+	go test -timeout 5m ./t/e2e/node/... -network=$(networkid) $(e2e_extraflags)
+	go test -timeout 50m ./t/e2e/jail/... -network=$(networkid) $(e2e_extraflags)
+	go test -timeout 20m ./t/e2e/rpc/... -network=$(networkid) $(e2e_extraflags)
+	go test -timeout 20m ./t/e2e/whisper/... -network=$(networkid) $(e2e_extraflags)
+	go test -timeout 10m ./t/e2e/transactions/... -network=$(networkid) $(e2e_extraflags)
 	# e2e_test tag is required to include some files from ./lib without _test suffix
-	go test -timeout 40m -tags e2e_test ./lib -network=$(networkid)
+	go test -timeout 40m -tags e2e_test ./lib -network=$(networkid) $(e2e_extraflags)
+
+race-check: e2e_extraflags=-race ##@tests Run e2e tests with -race flag
```

Comment:
a better place for my previous comment. why don't you want to use this flag for unit tests?

Reply:
Only because it requires a new target. If we want a new `test-unit-race` target, I can add it.

Reply:
@dshulyak or do you mean making the `race-check` target run both unit tests and e2e tests. That might actually make more sense.